third-party/cmake/BuildTreesitter.cmake

```diff
@@ -0,0 +1,16 @@
+ExternalProject_Add(tree-sitter
+  PREFIX ${DEPS_BUILD_DIR}
+  URL ${TREESITTER_URL}
+  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/tree-sitter
+  DOWNLOAD_COMMAND ${CMAKE_COMMAND}
+  -DPREFIX=${DEPS_BUILD_DIR}
+  -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/tree-sitter
+  -DURL=${TREESITTER_URL}
+  -DEXPECTED_SHA256=${TREESITTER_SHA256}
+  -DTARGET=tree-sitter
+  -DUSE_EXISTING_SRC_DIR=${USE_EXISTING_SRC_DIR}
+  -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
+  BUILD_IN_SOURCE 1
+  PATCH_COMMAND ""
+  CONFIGURE_COMMAND ""
+  INSTALL_COMMAND "PREFIX=${DEPS_INSTALL_DIR}" "make" "install")
```

Comment:
I don't know any other way to do it, but I am sure there is one.

Reply:
Just a heads-up: I don't think the empty commands are robust in cmake 3.18 -- can you try with that version?

Reply:
Empty command just does nothing, which is not what I want I guess

Reply:
That seems to have changed in 3.18, at least in some cases.

Reply:
(see #12676 )

Reply:
Never mind, I just tested with CMake 3.18 (macOS) and it works.